terraform/deployments/cluster-services/kube_prometheus_stack.tf

```diff
@@ -236,6 +237,35 @@ resource "helm_release" "kube_prometheus_stack" {
         }
       }]
     }
+    alertmanager = {
+      alertmanagerSpec = {
+        podAntiAffinity = "hard"
+        replicas        = var.kube_prometheus_stack_replica_count
+        topologySpreadConstraints = [{
```

Comment:
I think the pod antiaffinity and spreadcontraints are not needed, by default the EKS schedule spread the pods across the AZ. I introduced a new variable too (see here": https://github.com/alphagov/govuk-infrastructure/blob/c34054dd9ca45aff71fa19d5fb75f4fed345a4a7/terraform/deployments/cluster-services/variables.tf#L46) that you can set your replicas to.